core/src/main/scala/kafka/log/ProducerStateManager.scala

```diff
@@ -297,18 +283,26 @@ private[log] class ProducerAppendInfo(val topicPartition: TopicPartition,
     }
   }
 
+  private def checkCoordinatorEpoch(endTxnMarker: EndTransactionMarker, offset: Long): Unit = {
+    if (updatedEntry.coordinatorEpoch > endTxnMarker.coordinatorEpoch) {
+      if (origin == AppendOrigin.Leader) {
+        warn(s"Detected invalid coordinator epoch for producerId $producerId at " +
+          s"offset $offset in partition $topicPartition: ${endTxnMarker.coordinatorEpoch} " +
+          s"is older than previously known coordinator epoch ${updatedEntry.coordinatorEpoch}")
```

Comment:
Foo the case where we log, should we indicate that we're ignoring it and whether this is safe or the user has to take some action?

Reply:
The damage (if any) is sort of already done at this point. I will change the level to INFO. I debated even using DEBUG, but I think this case is rare enough that we'd want to be sure it makes it to the logs.